### DISCUSSION TITLE
Control `details_row` using widgets

### DISCUSSION THREAD

**Discussion by susanu** [Laravel-Backpack/community-forum#712] [Ideas]
Hi,

I have the following setup in my project which enables me to control the details row using widgets.

`details_row.blade.php`
```
<div class="m-t-10 m-b-10 p-l-10 p-r-10 p-t-10 p-b-10" bp-section="crud-details-row">
    <div class="row">
        <div class="col-md-12">
            @include(backpack_view('inc.widgets'), ['widgets' => app('widgets')->where('section', 'details_row')->toArray()])
        </div>
    </div>
</div>
<div class="clearfix"></div>
```
`column.blade.php` (a new widget type)
```
@php
    if(!isset($widget['column'])) {
        return;
    }

    $column = $crud->makeSureColumnHasNeededAttributes($widget['column']);

    $columnPaths = array_map(function($item) use ($column) {
        return $item.'.'.$column['type'];
    }, config('backpack.crud.view_namespaces.columns'));
@endphp
@includeFirst($columnPaths, ['entry' => $entry, 'column' => $column])
```

Configuration (`skills` being a json column casted as array in model)
```
Widget::add([
    'type' => 'column',
    'section' => 'details_row',
    'column' => [
        'name' => 'skills',
        'type' => 'table',
        'columns' => [
            'name' => 'Name',
        ]
    ],
]);
```

**Comment by pxpm** [comment]
Hey @susanu thanks for one more contribution 🙏 

Neat trick, let me ask you, what's the real intent behind this ? Is that you are then able to use columns in the details_row or you use other widget types there? 

Can you share some screenshot of what you've achieved using this solution ? 

I like it, I am not seeing a way to "generalize" it, at most what we could have in Backpack is changing the "blank details row template" to a "details row with widgets template" . 

I really like that concept, by default you could just `CRUD::set('list.detailsRow', true)'; Widget::add(...)` and it would work without having to change stuff in html. 

I've created https://github.com/Laravel-Backpack/CRUD/issues/5347 in the CRUD repository to discuss this with the team. 

Thanks again @susanu 🙏

**Comment by susanu** [comment]
Hi @pxpm,

Yes, the real intent here is to use columns in `details_row` since the data displayed there is related to the entry.

![image](https://github.com/Laravel-Backpack/community-forum/assets/17728443/cd9a825a-09f2-4359-8378-c4682b32c389)

In this example i have a one to many relationship called `accounts` in my model which stores multiple platform accounts for that model (skype, youtube, email, twitter and so on).
Instead of displaying that in the main table which will take a lot of space, i decided to display it on the details row which will show all accounts the user has in a table (see `relationship_table` column i proposed).
In another example instead of a table, i have tabs

![image](https://github.com/Laravel-Backpack/community-forum/assets/17728443/29c7a530-e0f3-4c2f-aaa2-ba1aa5bf3e78)

Yes, the details row above is composed using only columns, no custom html needed. I have the custom column types: `tabs`, `glue` (which combines 2 columns next to each other), `row` which will split columns in half.
Since you closed the pull request regarding the `relationship_table`, maybe it will be a good idea to display some kind of table with multiple attributes like `subfields` in the field config.
If, for example, set `subcolumns` into the `relationship` column configuration, a table will be displayed.

**Reply by pxpm** [reply]
> Since you closed the pull request regarding the relationship_table, maybe it will be a good idea to display some kind of table with multiple attributes like subfields in the field config.

I think this is already happening and it's possible to define the subfields in the relationship column.
Did you had any issue using it ? You just define `subfields` (yeah, name sucks 😞 🤷 ) in the column definition.

> Yes, the details row above is composed using only columns, no custom html needed. I have the custom column types: tabs, glue

What if instead of, e.g tabs, being a "column type", it was a widget type ? So it could accept like `tabTitle` and `tabContent` ? Wouldn't it make that widget more generic ?

**Reply by susanu** [reply]
I just tried setting up `subfields` in a `relationship` column and it works, thanks for the information.
I guess i missed it in the docs.

Regarding the `tabs` widget, i prefer creating new column types that i can use in different places (in show operation for example) and also i can use it in a widget with the `column` widget type.

Since most of the stuff is tied to the entry, i would stick with creating new column types and the widget just display those columns.

**Comment by pxpm** [comment]
Thanks for the suggestion @susanu It was merged into backpack/crud. A new version will be released later today.

You can start using your custom widgets on the details row. 👍 

Cheers
